src/basho_bench_duration.erl

```diff
@@ -0,0 +1,133 @@
+-module(basho_bench_duration).
+
+-behavior(gen_server).
+
+-export([
+    run/0,
+    remaining/0
+]).
+
+-export([start_link/0]).
+
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    terminate/2,
+    code_change/3
+]).
+
+-record(state, {
+    ref,
+    duration,
+    start
+}).
+
+-include("basho_bench.hrl").
+
+
+run() ->
+    gen_server:call(?MODULE, run).
+
+
+remaining() ->
+    gen_server:call(?MODULE, remaining).
+
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+
+init([]) ->
+    run_hook(basho_bench_config:get(pre_hook, no_op)),
+    Ref = erlang:monitor(process, whereis(basho_bench_run_sup)),
+    {ok, #state{ref=Ref}}.
+
+
+handle_call(run, _From, State) ->
+    DurationMins = basho_bench_config:get(duration, 1),
+    lager:info("Starting with duration: ~p", [DurationMins]),
+    NewState = State#state{
```

Comment:
lager:info("Starting with duration: ~p~n", [DurationMins]),   %% Needs ~n for new line